docs/eove-his-interface.md

```diff
@@ -0,0 +1,765 @@
+The client shall reply with special [pongs messages](#pong-command) before the timeout (typically 5s).
+
+Client reads:
+
+```json
+{
+  "type": "PING"
+}
+```
+
+Client writes:
+
+```json
+{
+  "type": "PONG"
+}
+```
+
+When client fails to send a pong message, server will consider client as disconnected.
+This means that client will have to enable [high level communication](#high-level-communication) again.
+
+### Data format
+
+When decoded and delimited, messages are JSON-valid strings.
+In other words every used types are compatible with JSON (number, string, array, etc.).
+
+Dates (with time component) are converted to epoch with a millisecond precision.
+
+## Client messages
+
+Client can send message to server and these messages are mostly commands.
+This means message type will be an order like `DO_SOMETHING` or `GET_ME_SOMETHING`.
+
+Commands will often receive a positive response message like `DO_SOMETHING_SUCCEEDED`.
+
+`DO_SOMETHING` or `GET_ME_SOMETHING` are not valid commands but are used in several examples because they are concise enough.
+
+A message has the following simplified type:
+
+```ts
+interface Message {
+  type: string;
+  reference?: string;
+  payload?: any;
+}
+```
+
+### High level communication
+
+By default server will ignore any message coming from client because high level communication is disabled.
+Client should send a [START_COMMUNICATION](#startcommunication-command) command to enable it.
+
+### Using a reference
+
+An optional `reference` field is available to help the client link messages and replies.
+The server returns the `reference` field provided unchanged in its replies.
+
+Client writes:
+
+```json
+{
+  "type": "GET_SOMETHING",
+  "reference": "1"
+}
+```
+
+As a response, client reads:
+
+```json
+{
+  "type": "GET_SOMETHING_SUCCEEDED",
+  "reference": "1"
+}
+```
+
+### START_COMMUNICATION command
+
+This command must be the first one because it enables high level communication with server.
+
+Client writes:
+
+```json
+{
+  "type": "START_COMMUNICATION"
+}
+```
+
+Client reads:
+
+```json
+{
+  "type": "START_COMMUNICATION_SUCCEEDED",
+  "payload": {
+    "apiVersion": "1.0.0"
+  }
+}
+```
+
+The succeeded response includes the HIS server API version.
+
+### STOP_COMMUNICATION command
+
+This command will properly stop high level communication with server.
+
+Client writes:
+
+```json
+{
+  "type": "STOP_COMMUNICATION"
+}
+```
+
+Client reads:
+
+```json
+{
+  "type": "STOP_COMMUNICATION_SUCCEEDED"
+}
+```
+
+No other command will be processed by server except another start communication one.
+
+### PONG command
+
+Client writes:
+
+```json
+{
+  "type": "PONG"
+}
+```
+
+This message won't receive a corresponding response but another ping message will follow.
+
+### SUBSCRIBE command
+
+Client can subscribe to various channels to receive data in real time.
+
+Example for a waveforms subscription:
+
+```json
+{
+  "type": "SUBSCRIBE",
+  "payload": ["waveforms"]
+}
+```
+
```

Comment:
I would add a multi-channel subscription example to show the payload type (array):

```json
{
  "type": "SUBSCRIBE",
  "payload": ["waveforms", "monitorings"]
}
```